src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -396,8 +408,69 @@ public List<Pet> getPetsFromSupplier(Supplier supplier) {
     }
 
     @Override
-    public void clearMasterList() {
+    public void clearCurrList() {
         filteredAll.clear();
+        filteredCurrList.clear();
+    }
+
+    @Override
+    public void switchToBuyerList() {
+        filteredCurrList = filteredBuyers;
+    }
+
+    @Override
+    public void switchToSupplierList() {
+        filteredCurrList = filteredSuppliers;
+    }
+
+    @Override
+    public void switchToDelivererList() {
+        filteredCurrList = filteredDeliverers;
+    }
+
+    @Override
+    public void switchToOrderList() {
+        filteredCurrList = filteredOrders;
+    }
+
+    @Override
+    public void switchToPetList() {
+        filteredCurrList = filteredPets;
+    }
+
+    @Override
+    public void switchToMainList() {
+        filteredCurrList = new FilteredList<>(filteredAll.getMasterList());
+    }
+
+    @Override
+    public void checkBuyerOrder(Buyer buyer) {
+        ObservableList<Order> orders = FXCollections.observableArrayList(getOrdersFromBuyer(buyer));
+        filteredCurrList = new FilteredList<>(orders);
+    }
+
+    @Override
+    public void checkSupplierPet(Supplier supplier) {
+        ObservableList<Pet> pets = FXCollections.observableArrayList(getPetsFromSupplier(supplier));
+        filteredCurrList = new FilteredList<>(pets);
+    }
+
+    @Override
+    public void checkDelivererOrder(Deliverer deliverer) {
```

Comment:
It is not necessary to check deliverer order I guess? Since for the current version we do not do any operations on the deliverers' order list. But it is ok to keep it.